src/lib/components/Widget.js

```diff
@@ -0,0 +1,262 @@
+            Authorization: "bearer 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",
+          },
+          body: JSON.stringify({
+            query: `{
+            rateLimit {
+              remaining
+            }
+            repository(name: "SSW.Rules.Content", owner: "SSWConsulting") {
+              object(expression: "main:${
+                list[counter + this.state.numberOfRules]
+              }") {
+                ... on Blob {
+                  text
+                }
+              }
+            }
+          }`,
+          }),
+        }).then((res) => res.json());
+        if (response.data.repository.object != null) {
+          returnList = [...returnList, response.data.repository.object.text];
+        } else {
+          returnList = [...returnList, null];
+        }
+        counter++;
+      }
+    }
+    return returnList;
+  }
+
+  async componentDidMount() {
+    this.determineTheme();
+    const pullRequests = await this.fetchPullRequests();
+    var retrievalList = [];
+    var rulesList = [];
+    for (let pr of pullRequests.data.search.nodes) {
+      for (let file of pr.files.nodes) {
+        if (
+          !retrievalList.includes(file.path) &&
+          file.path.substring(file.path.length - 3) === ".md" &&
+          file.path.substring(0, 6) === "rules/"
+        ) {
+          retrievalList = [...retrievalList, file.path];
+          rulesList = [
+            ...rulesList,
+            {
+              author: pr.author.login,
+              timestamp: new Date(pr.mergedAt),
+            },
+          ];
+        }
+      }
+    }
+    var files = await this.fetchFileContents(retrievalList);
+    for (let [i, file] of files.entries()) {
+      if (file != null) {
+        var title = this.extractFromRuleContent("title", file);
+        var uri = this.extractFromRuleContent("uri", file);
+        this.setState({
+          isLoaded: true,
+          rules: [
+            ...this.state.rules,
+            {
+              id: this.state.rules.length + 1,
+              uri: uri,
+              path: file.path,
+              title: title,
+              author: rulesList[i].author,
+              timestamp: rulesList[i].timestamp,
+            },
+          ],
+        });
+      }
+    }
+  }
+
+  determineTheme() {
+    if (this.state.isDarkMode === undefined) {
+      let browserDarkMode = window.matchMedia("(prefers-color-scheme: dark)");
+      this.setState({
+        isDarkMode: browserDarkMode ? true : false,
+      });
+    }
+  }
+
+  render() {
+    const { error, isLoaded, rules } = this.state;
+    return (
+      <div className="rules-widget-container">
+        <div className="rules-widget-title">
+          <a href="https://www.ssw.com.au/ssw">
+            <img src={Logo} alt="SSW Logo" height="60" width="130"></img>
+          </a>
+          <h1>
+            <a
+              rel="noreferrer"
+              target="_blank"
+              href="https://www.ssw.com.au/rules/"
```

Comment:
Can this URL be pulled from an environment variable? What if the rules URL changes again?